### PR TITLE
Moved Armament, Attack, Move, Air features and dependencies to Mods.Common

### PR DESCRIPTION
**Pull request by reaperrr**
Sorry for this monster of a commit, but the code is too entangled to break it into smaller chunks (unless you can live with non-compileable commits).

Applied StyleCop fixes, extracted Exit trait into its own file, tested in-game for any noticable regressions, didn't notice anything out of the ordinary.

### CODE REVIEW

File: OpenRA.Mods.Common/Traits/Mobile.cs [hunk 1]
```diff
@@ -250,60 +249,60 @@ public SubCell GetAvailableSubCell(World world, Actor self, CPos cell, SubCell p
...
-		public readonly Actor self;
+		public readonly Actor Self;
 		public readonly MobileInfo Info;
 		public bool IsMoving { get; set; }
 
-		int __facing;
-		CPos __fromCell, __toCell;
-		public SubCell fromSubCell, toSubCell;
+		int facing;
+		public CPos FromCell1, ToCell1;
```
Comment (Mailaender) [inline OpenRA.Mods.Common/Traits/Mobile.cs]: This is never used anywhere else so it can stay private.

Comment (Mailaender) [inline OpenRA.Mods.Common/Traits/Mobile.cs]: You can then also avoid enumerating the variables.

**Comment by Mailaender** [comment]
> Sorry for this monster of a commit, but the code is too entangled to break it into smaller chunks (unless you can live with non-compileable commits).

It is fine. The size is still reviewable. Let's avoid non-compileable commits at all costs.

**Comment by reaperrr** [comment]
> It is fine. The size is still reviewable. Let's avoid non-compileable commits at all costs.

OK. This is probably the last chunk of this size anyway. Building, Move, Attack, Air and Armament-related stuff were the parts with the most inter-dependencies, the rest of Mods.RA should be much easier to move in smaller portions.

**Comment by Phrohdoh** [comment]
Armament's `Self| doesn't need to have an explicit `this` (any other similar situations can be fixed too).

File: OpenRA.Mods.Common/Traits/Armament.cs [hunk 2]
```diff
@@ -75,11 +74,11 @@ public class Armament : UpgradableTrait<ArmamentInfo>, ITick, IExplodeModifier
 		public Armament(Actor self, ArmamentInfo info)
 			: base(info)
 		{
-			this.self = self;
+			this.Self = self;
```
Comment (Mailaender) [inline OpenRA.Mods.Common/Traits/Armament.cs]: I can only find this one.

File: OpenRA.Mods.Common/Traits/Mobile.cs [hunk 3]
```diff
@@ -250,60 +249,60 @@ public SubCell GetAvailableSubCell(World world, Actor self, CPos cell, SubCell p
...
-		const int spreadTicksBeforePathing = 5;
-		internal int ticksBeforePathing = 0;
+		const int AverageTicksBeforePathing = 5;
+		const int SpreadTicksBeforePathing = 5;
+		internal int TicksBeforePathing = 0;
 
 		public Mobile(ActorInitializer init, MobileInfo info)
 		{
-			this.self = init.self;
+			this.Self = init.self;
```
Comment (Mailaender) [inline OpenRA.Mods.Common/Traits/Mobile.cs]: As well as this one.

File: OpenRA.Mods.Common/Traits/Mobile.cs [hunk 4]
```diff
@@ -250,60 +249,60 @@ public SubCell GetAvailableSubCell(World world, Actor self, CPos cell, SubCell p
 
 	public class Mobile : IIssueOrder, IResolveOrder, IOrderVoice, IPositionable, IMove, IFacing, ISync, INotifyAddedToWorld, INotifyRemovedFromWorld, INotifyBlockingMove
 	{
-		public readonly Actor self;
+		public readonly Actor Self;
```
Comment (pchote) [inline OpenRA.Mods.Common/Traits/Mobile.cs]: Does anything else actually reference this?  It would be better to keep this private and lower case.

Comment (Mailaender) [inline OpenRA.Mods.Common/Traits/Mobile.cs]: No, you are right. The `public` can safely be removed.

**Comment by reaperrr** [comment]
Adressed what was pointed out, but give Mobile in particular another look to make sure I didn't misunderstand anything, StyleCop also asked me to move up constants after the suggested changes.

File: OpenRA.Mods.Common/Traits/Mobile.cs [hunk 5]
```diff
@@ -250,60 +249,60 @@ public SubCell GetAvailableSubCell(World world, Actor self, CPos cell, SubCell p
...
 
-		const int avgTicksBeforePathing = 5;
-		const int spreadTicksBeforePathing = 5;
-		internal int ticksBeforePathing = 0;
-
 		public Mobile(ActorInitializer init, MobileInfo info)
 		{
-			this.self = init.self;
+			self = init.self;
 			this.Info = info;
```
Comment (Mailaender) [inline OpenRA.Mods.Common/Traits/Mobile.cs]: While you are at it you can also remove this `this.`

File: OpenRA.Mods.Common/Activities/Move/Move.cs [hunk 6]
```diff
@@ -291,42 +290,42 @@ public override IEnumerable<Target> GetTargets(Actor self)
...
-				this.move = move;
-				this.from = from;
-				this.to = to;
-				this.fromFacing = fromFacing;
-				this.toFacing = toFacing;
+				this.Move = move;
+				this.From = from;
+				this.To = to;
+				this.FromFacing = fromFacing;
+				this.ToFacing = toFacing;
```
Comment (Mailaender) [inline OpenRA.Mods.Common/Activities/Move/Move.cs]: Another bunch of `this.` marked for removal.

File: OpenRA.Mods.Common/Activities/Move/Move.cs [hunk 7]
```diff
@@ -291,42 +290,42 @@ public override IEnumerable<Target> GetTargets(Actor self)
...
 				this.moveFraction = startingFraction;
```
Comment (Mailaender) [inline OpenRA.Mods.Common/Activities/Move/Move.cs]: Same here

File: OpenRA.Mods.Common/Activities/Move/Move.cs [hunk 8]
```diff
@@ -291,42 +290,42 @@ public override IEnumerable<Target> GetTargets(Actor self)
...
-				this.moveFractionTotal = (to - from).Length;
+				this.MoveFractionTotal = (to - from).Length;
```
Comment (Mailaender) [inline OpenRA.Mods.Common/Activities/Move/Move.cs]: and here.

File: OpenRA.Mods.Common/Activities/Move/MoveAdjacentTo.cs [hunk 9]
```diff
@@ -26,14 +26,14 @@ public class MoveAdjacentTo : Activity
...
-		protected Target target { get; private set; }
+		protected Target Target { get; private set; }
 		protected CPos targetPosition;
 		Activity inner;
 		bool repath;
 
 		public MoveAdjacentTo(Actor self, Target target)
 		{
-			this.target = target;
+			this.Target = target;
```
Comment (Mailaender) [inline OpenRA.Mods.Common/Activities/Move/MoveAdjacentTo.cs]: Another hit.

Comment (reaperrr) [inline OpenRA.Mods.Common/Activities/Move/MoveAdjacentTo.cs]: OK.
Generally speaking, is there an easier way than trial and error to spot those cases? Would save both me and reviewers some hassle in the future.

**Comment by reaperrr** [comment]
Updated.

File: OpenRA.Mods.Common/Activities/Move/MoveAdjacentTo.cs (hunk 9 above)
Comment (Mailaender) [inline OpenRA.Mods.Common/Activities/Move/MoveAdjacentTo.cs]: I self reviewed my https://github.com/OpenRA/OpenRA/pull/7268 on GitHub which gives a good overview.

**Comment by Mailaender** [comment]
:+1: This is looking good. Those files are definitely common RTS elements. Thanks for the cleanup.

**Comment by Phrohdoh** [comment]
Looks solid to me. :+1:

**Pull request merged**